components/iostreams/CMakeLists.txt

```diff
@@ -38,9 +38,15 @@ add_hpx_component(iostreams
   SOURCES ${iostreams_sources}
   ${_exclude_from_all_flag})
 
+if (TARGET hpx::apex)
+  target_link_libraries(iostreams_component PRIVATE hpx::apex)
+endif()
+
 target_compile_definitions(iostreams_component
   PRIVATE HPX_IOSTREAMS_MODULE_EXPORTS)
 
+target_link_libraries(iostreams_component PRIVATE hpx::boost)
```

Comment:
Wouldn't it be easier to make Boost into a PUBLIC dependency of HPX core? This would also enforce that any dependent code links with the same Boost version as HPX was built against.

Reply:
HPX actually already link to boost but the components are independent so we also have to link them if I understood well

Reply:
Are they independent? They do need functionality from the main HPX library and link to it as well.

Reply:
Could you show me where it is linked to the main hpx lib ?